src/batch_norm.jl

```diff
@@ -0,0 +1,118 @@
+# Code adapted from Flux.jl:
+# https://github.com/FluxML/Flux.jl/blob/68ba6e4e2fa4b86e2fef8dc6d0a5d795428a6fac/src/layers/normalise.jl#L117-L206
+# License: https://github.com/FluxML/Flux.jl/blob/master/LICENSE.md
+
+"""
+InvertibleBatchNorm(β, logγ, μ, σ², ϵ::AbstractFloat, momentum::AbstractFloat, active::Bool)
+
+β, logγ - learned parameters
+
+`InvertibleBatchNorm` computes the mean and variance for each each `NxD` slice and
+shifts them to have a new mean and variance (corresponding to the learnable,
+per-channel `bias` and `scale` parameters). 
+
+See [Batch Normalization: Accelerating Deep Network Training by Reducing
+Internal Covariate Shift](https://arxiv.org/pdf/1502.03167.pdf).
+
+"""
+mutable struct InvertibleBatchNorm{T1,T2,T3} <: Bijector
+    β::T1
+    logγ::T1
+    μ::T2  # moving mean
+    σ²::T2 # moving st
+    ϵ::T3
+    momentum::T3
+    active::Bool # true when training
+end
+
+InvertibleBatchNorm(dims::Int,
+                    container=Array; 
+                    ϵ::AbstractFloat=1f-5,
+                    momentum::AbstractFloat=0.1f0) = (
+                    InvertibleBatchNorm(
+                            container(zeros(dims, 1)),
+                            container(zeros(dims, 1)),
+                            zeros(dims, 1),
+                            ones(dims, 1),
+                            ϵ,
+                            momentum,
+                            true
+                    ))
+
+function affinesize(x)
+    dims = length(size(x))
+    channels = size(x, dims)
+    affinesize = ones(Int, dims)
+    affinesize[end-1] = channels
+    return affinesize
+end
+
+logabsdetjac(
+    t::T,
+    x
+) where{T<:InvertibleBatchNorm} = forward(t, x).logabsdetjac
+
+function _compute_μ_σ²(t::InvertibleBatchNorm, x)
+    @assert size(x, ndims(x)) == 
```

Comment:
Style: this assertion is quite hard to read. Maybe consider the functional macro style:
```julia
@assert(
    size(x, ndims(x)) == length(t.μ),
    "`InvertibleBatchNorm` expected $(length(t.μ)) channels, got $(size(x,ndims(x)))"
)
```